### ISSUE TITLE
HighLevelInterface.MixintprogSolution untyped

### ISSUE THREAD

**Issue by mbesancon** [JuliaOpt/MathProgBase.jl#210]
Is there a reason why MixintprogSolution is untyped? I was expecting status to always be a symbol (and leveraging this on top of MPBase).

```julia
help?> MathProgBase.HighLevelInterface.MixintprogSolution
  No documentation found.

  Summary:

  mutable struct MathProgBase.HighLevelInterface.MixintprogSolution <: Any

  Fields:

  status :: Any
  objval :: Any
  sol    :: Any
  attrs  :: Any
```

**Comment by mbesancon** [comment]
Yes I read about [MathOptInterface](https://discourse.julialang.org/t/mathoptinterface-and-upcoming-breaking-changes-in-jump-0-19/4874/2), but Miles seemed to imply that this was coming in an uncertain future. 

We are using MathProgBase instead of JuMP because we didn't want the modeling overhead: we try to build a generic column generation framework.

**Comment by odow** [comment]
> Miles seemed to imply that this was coming in an uncertain future.

The MOI train is gaining steam. JuMP is now working and a number of solvers have been, or are close to being, ported.

Column generation is one notable exception that we haven't experimented with much.

The current expectation is that you would add a variable, and then call a [MultirowChange](http://www.juliaopt.org/MathOptInterface.jl/latest/apireference.html#MathOptInterface.MultirowChange).  (See also https://github.com/JuliaOpt/MathOptInterface.jl/issues/360.)

If you have some thoughts on a nice, generic column generation interface, now is a great time to get on-board with suggestions. The API is still going through some changes, but in the near future it will likely settle down.

> I read the commit you linked to, it does not seem to indicate why the fields are untyped

It was written five years ago in one of the first commits when the majority of the interface wasn't written and hasn't been touched since :) So I would say there is no reason...

**Comment by mbesancon** [comment]
Thanks for the information. So you would confirm that it would be better to plug a column generation framework on `MathOptInterface` rather than MPBase or JuMP?

We are thinking about a way to make it flexible enough but with the basic use case covered. The working repo is [Linda](https://github.com/ds4dm/Linda.jl), glad to have comments :)

We're still adjusting the core interface, but will have to go through obvious adjustments for MathOptInt

**Comment by odow** [comment]
>  So you would confirm that it would be better to plug a column generation framework on MathOptInterface rather than MPBase or JuMP?

:100: 

> We are thinking about a way to make it flexible enough but with the basic use case covered

MOI should make this a lot easier than MPB. It has much better support for solution statuses so you won't have to resort to hacks like [this](https://github.com/ds4dm/Linda.jl/blob/e4ca6e082514ba9799e2370f59a49174722f5de3/src/problem_status.jl#L72), and has a much more solid way of querying the primal and dual solutions, and getting information about the availability of any such solutions. It is also more generic than MPB and can support a wider variety of problems.

p.s., If you are not aware, we are holding a JuMP developers workshop next month in Bordeaux. You are most welcome to attend http://www.juliaopt.org/meetings/bordeaux2018/.

**Comment by mbesancon** [comment]
thanks a lot for the info. Unfortunately, I won't be able to attend, I'll come back to Europe two days after the end of the workshop but it would really have been great.

Regarding the statuses, what we did is clearly way too hacky as you noticed. Symbols might be too error-prone for statuses, but we were thinking users may need new status types to define new Master and SubProblems, so I'm not sure enum is the way to go

**Comment by odow** [comment]
>  we were thinking users may need new status types to define new Master and SubProblems

Have a read of 
http://www.juliaopt.org/MathOptInterface.jl/latest/apireference.html#Termination-Status-1
http://www.juliaopt.org/MathOptInterface.jl/latest/apireference.html#Result-Status-1
Can you think of a situation that isn't covered by these statuses?

**Comment by mlubin** [comment]
It's a common meme at Google that the way that works is deprecated and the recommended way is still under construction :)
